apps/next/src/memlab/lib/analyzer.ts

```diff
@@ -0,0 +1,750 @@
+ * of the License at https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import * as fs from 'node:fs';
+import * as path from 'node:path';
+import memlabApiDefault, { ConsoleMode, findLeaks } from '@memlab/api';
+import { designToolkitFilter } from '../filters';
+import {
+  compareWithBaseline,
+  formatBaselineComparison,
+  getGitCommitHash,
+  loadBaseline,
+  saveBaseline,
+  updateBaseline,
+} from './baseline';
+import { DEFAULT_MAX_RETAINED_SIZE, LEAK_PATTERNS } from './constants';
+import { formatBytes } from './heap-snapshot';
+import { validateThresholdJson } from './threshold-validator';
+import { TimeoutError, withTimeout } from './timeout';
+import type {
+  AllocationLocation,
+  AnalysisOptions,
+  AnalysisResult,
+  ComponentThreshold,
+  LeakInfo,
+  MemlabReport,
+  RetainerPathItem,
+  ThresholdConfig,
+} from './types';
+
+/**
+ * Default threshold configuration
+ */
+const DEFAULT_THRESHOLD: ComponentThreshold = {
+  maxLeakedObjects: 0,
+  maxRetainedSize: DEFAULT_MAX_RETAINED_SIZE,
+};
+
+// =============================================================================
+// THRESHOLD CACHING
+// =============================================================================
+
+/** Cached threshold configuration */
+let cachedThresholds: ThresholdConfig | null = null;
+
+/** Path of the cached threshold file */
+let cachedThresholdsPath: string | null = null;
+
+/**
+ * Get thresholds from cache or load from file
+ *
+ * @param thresholdFile - Path to threshold configuration file
+ * @returns Threshold configuration (from cache if available)
+ */
+function getCachedThresholds(thresholdFile: string): ThresholdConfig {
+  // Return cached if same file
+  if (cachedThresholds && cachedThresholdsPath === thresholdFile) {
+    return cachedThresholds;
+  }
+
+  // Load and validate
+  if (!fs.existsSync(thresholdFile)) {
+    return {
+      global: DEFAULT_THRESHOLD,
+      components: {},
+    };
+  }
+
+  try {
+    const content = fs.readFileSync(thresholdFile, 'utf-8');
+    const validation = validateThresholdJson(content);
+
+    if (!validation.valid) {
+      console.warn('⚠️ Threshold config validation errors:');
+      for (const error of validation.errors) {
+        console.warn(`   ❌ ${error}`);
+      }
+      return {
+        global: DEFAULT_THRESHOLD,
+        components: {},
+      };
+    }
+
+    // Log warnings but continue
+    if (validation.warnings.length > 0) {
+      for (const warning of validation.warnings) {
+        console.warn(`   ⚠️ ${warning}`);
+      }
+    }
+
+    const parsed: ThresholdConfig = JSON.parse(content);
```

Comment:
validateThresholdJson parses the json and then we parse it again on line 100. It'd be more efficient to return the parsed object from the validate func to reuse on line 100